supplementary/alica_ros1/alica_tests/src/test/test_blackboard.cpp

```diff
@@ -117,5 +129,638 @@ TEST_F(TestBlackBoard, testMapping)
+
+    bb_locked.set<UnknownType>("value", UnknownType());
+    EXPECT_EQ(typeid(UnknownType), typeid(bb_locked.get<UnknownType>("value")));
+}
+
+TEST_F(TestBlackboard, testSetWithDifferentTypeNotInPML)
+{
+    Blackboard bb;
+    auto bb_locked = LockedBlackboardRW(bb);
+    bb_locked.set<int64_t>("value", 1);
+    EXPECT_EQ(bb_locked.get<int64_t>("value"), 1);
+
+    bb_locked.set<double>("value", 2.0);
+    EXPECT_EQ(bb_locked.get<double>("value"), 2.0);
+}
+
+TEST_F(TestBlackboard, testSetAnyWithDifferentType)
+{
+    Blackboard bb;
+    auto bb_locked = LockedBlackboardRW(bb);
+    bb_locked.set<alica::PlanStatus>("value", alica::PlanStatus::Success);
+    EXPECT_EQ(bb_locked.get<alica::PlanStatus>("value"), alica::PlanStatus::Success);
+
+    bb_locked.set<UnknownType>("value", UnknownType());
+    EXPECT_EQ(typeid(UnknownType), typeid(bb_locked.get<UnknownType>("value")));
+
+    bb_locked.set<int64_t>("value", 2);
+    EXPECT_EQ(bb_locked.get<int64_t>("value"), 2);
+}
+
+TEST_F(TestBlackboard, testInitWithUnsupportedType)
+{
+    std::unique_ptr<alica::BlackboardBlueprint> blueprint = std::make_unique<alica::BlackboardBlueprint>();
+    blueprint->addKey("unknown", "UnknownType", std::nullopt);
+    EXPECT_THROW(
+            {
+                try {
+                    alica::Blackboard(blueprint.get());
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(), "Blackboard exception: key: unknown has an unsupported type");
+                    throw;
+                }
+            },
+            BlackboardException);
+}
+
+TEST_F(TestBlackboard, testInitWithDefaultValueForAny)
+{
+    std::unique_ptr<alica::BlackboardBlueprint> blueprint = std::make_unique<alica::BlackboardBlueprint>();
+    blueprint->addKey("anyWithDefault", "std::any", "defaultValue");
+    EXPECT_THROW(
+            {
+                try {
+                    alica::Blackboard(blueprint.get());
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(), "Blackboard exception: key: anyWithDefault cannot have a default value because it is of type std::any");
+                    throw;
+                }
+            },
+            BlackboardException);
+}
+
+TEST_F(TestBlackboard, testInitWithWrongDefaultValueType)
+{
+    std::unique_ptr<alica::BlackboardBlueprint> blueprint = std::make_unique<alica::BlackboardBlueprint>();
+    blueprint->addKey("anyWithDefault", "int64", "defaultValue");
+    EXPECT_THROW(
+            {
+                try {
+                    alica::Blackboard(blueprint.get());
+                } catch (const BlackboardException& e) {
+                    EXPECT_STREQ(e.what(), "Blackboard exception: Could not set key: anyWithDefault, from default value: defaultValue, type: int64, details: "
+                                           "yaml-cpp: error at line 1, column 1: bad conversion");
```

Comment:
this can become flaky really soon. Can't rely on the exception msg thrown by yaml cpp library. Best not to check the excpetion msg. Should suffice to just check if an exception is thrown